lib/cJSON.c

```diff
@@ -2144,9 +2143,19 @@ CJSON_PUBLIC(cJSON *) snowflake_cJSON_DetachItemFromObjectCaseSensitive(cJSON *o
     return snowflake_cJSON_DetachItemViaPointer(object, to_detach);
 }
 
-CJSON_PUBLIC(void) snowflake_cJSON_DeleteItemFromObject(cJSON *object, const char *string)
+//delete the key word, if recurse = true, it will delete any children contain the key word
+CJSON_PUBLIC(void) snowflake_cJSON_DeleteItemFromObject(cJSON *object, const char *string, cJSON_bool recurse)
 {
     snowflake_cJSON_Delete(snowflake_cJSON_DetachItemFromObject(object, string));
+    if(recurse){
```

Comment:
I don't quite follow, can you please explain what this method used to do and what's doing now (maybe add it as a comment)

Reply:
Sure, this function could help us delete some attributes in the object, for example, json = {"data":XXX,"rowset":XXX}, if we call this function, snowflake_cJSON_Delete(json, "rowset"), it could help us delete the "rowset" and return {"data":XXX}. And for recurse, it helps to handle the case {"data":{"type":XXX, "rowset":XXX}}.